sql/core/src/main/scala/org/apache/spark/sql/internal/BaseSessionStateBuilder.scala

```diff
@@ -175,6 +175,7 @@ abstract class BaseSessionStateBuilder(
     override val postHocResolutionRules: Seq[Rule[LogicalPlan]] =
       PreprocessTableCreation(session) +:
         PreprocessTableInsertion(conf) +:
+        ResolveUpCast +:
```

Comment:
Write a comment and explain this rule is to resolve the upcast added in PreprocessTableInsertion?